src/test/groovy/pl/futurecollars/invoicing/controller/InvoiceControllerStepwiseTest.groovy

```diff
@@ -0,0 +1,153 @@
+package pl.futurecollars.invoicing.controller
+
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc
+import org.springframework.boot.test.context.SpringBootTest
+import org.springframework.http.MediaType
+import org.springframework.test.web.servlet.MockMvc
+import pl.futurecollars.invoicing.Helpers.TestHelpers
+import pl.futurecollars.invoicing.model.Invoice
+import pl.futurecollars.invoicing.utils.JsonService
+import spock.lang.Specification
+import spock.lang.Stepwise
+
+import java.time.LocalDate
+
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status
+
+@SpringBootTest
+@AutoConfigureMockMvc
+@Stepwise
+
+
+class InvoiceControllerStepwiseTest extends Specification {
+
+    @Autowired
+    private MockMvc mockMvc
+
+    @Autowired
+    private JsonService jsonService
+
+    private Invoice originalInvoice = TestHelpers.invoice(1)
+
+    private LocalDate updatedDate = LocalDate.of(2020, 02, 28)
+
+    def "get all invoices returns empty array when no invoices added"() {
+
+        when:
+        def response = mockMvc.perform(get("/invoices"))
```

Comment:
tutaj analogicznie do InvoiceControllerTest można wyciągnąć endpoint to stałej:
    private static final String ENDPOINT = "/invoices"